aten/src/ATen/core/type.cpp

```diff
@@ -237,6 +237,15 @@ ListTypePtr ListType::ofStrings() {
   return value;
 }
 
+TypePtr OptionalType::get(TypePtr inner) {
+  static ska::flat_hash_map<TypePtr, TypePtr> optionalTypePtrs;
+  if (optionalTypePtrs.find(inner) == optionalTypePtrs.end()) {
+    TypePtr t = TypeFactory::create<OptionalType>(inner);
+    optionalTypePtrs.emplace(inner, t);
+  }
```

Comment:
do you need locking here?

Reply:
yep 😬 will add (and it shouldn't matter if it's slow since it's behind a static singleton)

Reply:
update - locking fixed the bug described above lol. With the lock, this now works:
```
static const auto& call() {
    static auto inner_type = getTypePtr_<T>::call();
    static auto type = OptionalType::get(inner_type);
    return type;
}
```

I'm going to try to update the other container types to do something similar.

Reply:
same as above -- move & be careful not to use after move in the return statement